src/groupeddataframe/groupeddataframe.jl

```diff
@@ -429,8 +429,16 @@ end
 Base.axes(gd::GroupedDataFrame, i::Integer) = Base.OneTo(size(gd, i))
 
 Base.first(gd::GroupedDataFrame) = gd[1]
-Base.last(gd::GroupedDataFrame) = gd[end]
+function Base.first(gd::GroupedDataFrame, n::Integer)
+    n < 0 && throw(ArgumentError("Number of elements must be nonnegative"))
+    return @inbounds gd[1:min(n, end)]
```

Comment:
I would remove the `@inbounds` as the time saved by it is negligible and without it the code is safer:
```suggestion
    return gd[1:min(n, end)]
```